util/topsql/reporter/reporter.go

```diff
@@ -259,45 +254,50 @@ func (d *dataPoints) merge(other *dataPoints) {
 		return
 	}
 	if other.isInvalid() {
-		logutil.BgLogger().Warn("[top-sql] data points is invalid, it should never happen", zap.Reflect("self", *d), zap.Reflect("other", *other))
+		logutil.BgLogger().Warn("[top-sql] data points is invalid, it should never happen", zap.Any("self", d), zap.Any("other", other))
 		return
 	}
-	if !sort.IsSorted(other) {
-		// This situation appears only when time jump backward.
-		// If so, sort the data points by timestamp.
-		sort.Sort(other)
-	}
+	// Sort the dataPoints by timestamp to fix the affect of time jump backward.
+	sort.Sort(d)
```

Comment:
is `d` already sorted?

Reply:
No, d doesn't make sure is sorted.

Reply:
Why isn't `d` sorted? Seems that `d` only comes from 2 places:

1. `d` is the `others` below, `d` is empty so that it is sorted.

```
	if others == nil {
		others = &dataPoints{}
	}
	others.merge(evict)
	return others
```

2. `d` is the created from `d.merge`.

`d.merge` produce a sorted timestamp list so that it is sorted?

Reply:
3. `d`(`others`) can be create from `addEvictedCPUTime`.